### PR TITLE
hv: vm: properly reset pCPUs with LAPIC PT enabled during VM shutdown/reset

### PR DESCRIPTION
**Pull request by peterfang**
When a VM is configured with LAPIC PT mode and its vCPU is in x2APIC
mode, the corresponding pCPU needs to be reset during VM shutdown/reset
as its physical LAPIC was used by its guest

This commit fixes an issue where this reset never happens.
is_lapic_pt_enabled() needs to be called before reset_vcpu() to be able
to correctly reflect a vCPU's APIC mode.

A vCPU with LAPIC PT mode but in xAPIC mode does not require such reset,
since its physical LAPIC was not touched by its guest directly.

Tracked-On: #3708
Signed-off-by: Peter Fang <peter.fang@intel.com>
Reviewed-by: Eddie Dong <eddie.dong@intel.com>
Reviewed-by: Jack Ren <jack.ren@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
OK to verify

**Comment by acrnsi** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
5 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:sharing_config filename:/hypervisor/arch/x86/guest/vm.c func:shutdown_vm offset:548<br> reason:Function return value potentially unused. : this_pcpu_id<br><2> config:part_config filename:/hypervisor/arch/x86/guest/vm.c func:shutdown_vm offset:548<br> reason:Function return value potentially unused. : this_pcpu_id<br><3> config:sharing_config filename:/hypervisor/arch/x86/guest/vm.c func:reset_vm offset:632<br> reason:Function return value potentially unused. : this_pcpu_id<br><4> config:part_config filename:/hypervisor/arch/x86/guest/vm.c func:reset_vm offset:632<br> reason:Function return value potentially unused. : this_pcpu_id<br><5> config:common filename:/hypervisor/arch/x86/guest/vm.c func:reset_vm offset:657<br> reason:Logical conjunctions need brackets.<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
Have confrmed with huihuang, this check can be pass

File: hypervisor/arch/x86/guest/vm.c [hunk 1]
```diff
@@ -621,17 +629,39 @@ void start_vm(struct acrn_vm *vm)
...
+					pr_warn("%s: cannot offline self(%u)",
+						__func__, this_pcpu_id);
+					ret = -EINVAL;
+				}
+			}
+		}
+
+		wait_pcpus_offline(mask);
+
+		if (mask != 0UL && !start_pcpus(mask)) {
```
Comment (HuiHuangShi) [inline hypervisor/arch/x86/guest/vm.c]: please fix it by add brackets in another patch.
"**if ((mask != 0UL) && !start_pcpus(mask))** {"

Comment (peterfang) [reply hypervisor/arch/x86/guest/vm.c]: #3716

**Pull request merged**